structs-in-rust/article.md

```diff
@@ -0,0 +1,366 @@
+```
+struct A;
+```
+Don't be afraid if you see such strange struct definitions. You may wonder what is the purpose of unit structs when they don't contain any information. In reality, they are being used (see [std::fmt::Error](https://doc.rust-lang.org/std/fmt/struct.Error.html), [chrono::offset::Utc](https://docs.rs/chrono/0.4.19/chrono/offset/struct.Utc.html)), as they can have methods and different trait implementations. Both things I just mentioned will be discussed further in the article.
+
+### Tuple struct
+Fields of a struct don't always need to be named. A tuple struct is an alternative method of grouping multiple values together. You can define tuple structs like so:
+```
+//     [1]   [2]  [3]
+struct Point(i32, i32);
+
+// [1]: Struct name
+// [2]: first component / field of type i32
+// [3]: second component of type i32
+```
+### Struct initialization
+We already know how to define a struct. Let's write down previous examples.
+
+```
+// Ordinary struct
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+// Tuple struct
+struct PointTuple(i32, i32);
+```
+
+Now let's create some objects of type `Point` and `PointTuple`, i.e. instances of our structs.
+
+```
+//             [1]
+let p_named = Point {
+// [2] [3]
+    x: 13,
+    y: 37,
+};
+
+// [1]: struct name
+// [2]: field name
+// [3]: field value
+
+//               [1]       [2] [3]
+let p_unnamed = PointTuple(13, 37);
+
+// [1]: struct name
+// [2], [3]: field values
+```
+
+As you can see, the syntax of initialization is pretty straightforward. It is very similar to defining a JSON object, but with keys replaced with field names.
+
+Two things to keep in mind:
+- All fields must be assigned with values (there are no default parameters in Rust).
+- Order in which you list `key: value` pairs doesn't matter. You can do `let strange_order = Point { y: 37, x: 13 };` if you wish to.
+
+### Field access
+#### Get
+You can get the desired value using well-known dot notation.
+
+```
+let x = p_named.x;
+let y = p_named.y;
+```
+
+One may wonder, what is the syntax to address a specific field of a tuple struct. Fortunately, Rust has chosen the simplest way possible, by enumerating tuple components from *0* to *n*.
+
+```
+let x = p_unnamed.0;
+let y = p_unnamed.1;
+```
+#### Set
+It is possible to change the value of a field using the same dot notation, but the **struct instance must be defined as mutable**.
+```
+let mut p = PointTuple(1, 2);
+//  ^^^
+p.0 = 1000;
+assert_eq!(p.0, 1000);
+```
+Unlike some other languages, in Rust you cannot mark a specific field of a struct as mutable or immutable:
+
+```
+struct Struct {
+    mutable: mut i32,
+//           ^^^
+//           Expected type, found keyword `mut`.
+    immutable: bool,
+}
+```
+There are two options:
+- Have an immutable binding to the struct. Then you cannot change any of its fields.
+- Have a mutable binding, in which case you can set whatever field you choose.
+
+### Convenient field init syntax
```

Comment:
Wait, how does this differ from regular initialization?

Reply:
Ah, I see, misleading header. I made a change in the latest commit which should fix the problem